src/App.tsx

```diff
@@ -1,26 +1,210 @@
+      default:
+        return preperedTodos;
+    }
+  }
+
+  return preperedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<Todo[]>([]);
+  const [edit, setEdit] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const temp = (currentTodo: Todo) => {
+    setLoadingTodos(prevArray => [...prevArray, currentTodo]);
+  };
+
+  const filteredTodos = getTodosByStatus(status, todos);
+
+  const addTodo = async (newTodoTitle: string) => {
+    const editedTitle = newTodoTitle.trim();
+
+    if (!editedTitle) {
+      setErrorMessage('Title should not be empty');
+      wait(3000).finally(() => setErrorMessage('Title should not be empty'));
+
+      return;
+    } else {
+      setTempTodo({
+        id: 0,
+        userId: 839,
+        title: editedTitle,
+        completed: false,
+      });
+
+      return todosFromServer
+        .createTodos({
+          userId: 839,
+          title: editedTitle,
+          completed: false,
+        })
+        .then(newTodo => {
+          setTodos(prevTodos => [...prevTodos, newTodo]);
+          setTempTodo(null);
+        })
+        .catch(error => {
+          setErrorMessage('Unable to add a todo');
+          setTempTodo(null);
+          wait(3000).finally(() => setErrorMessage(''));
+          throw error;
+        });
+    }
+  };
+
+  const updateTodo = async (updatedTodo: Todo): Promise<void> => {
+    return todosFromServer
+      .updateTodos(updatedTodo)
+      .then((todo: Todo) => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(
+            thisTodo => thisTodo.id === updatedTodo.id,
+          );
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+        setEdit(false);
+      })
+      .catch(error => {
+        setEdit(true);
+        setErrorMessage('Unable to update a todo');
+        wait(3000).finally(() => setErrorMessage(''));
+        setLoadingTodos([]);
+        throw error;
+      });
+  };
+
+  const deleteTodo = (paramTodo: Todo) => {
+    todosFromServer
+      .deleteTodos(paramTodo.id)
+      .then(() =>
+        setTodos(prevTodos =>
+          prevTodos.filter(todo => todo.id !== paramTodo.id),
+        ),
+      )
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        wait(3000).finally(() => {
+          setErrorMessage('');
+        });
+        setLoadingTodos(array => array.filter(a => a.id !== paramTodo.id));
+      });
+  };
+
+  useEffect(() => {
+    todosFromServer
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+    wait(3000).finally(() => setErrorMessage(''));
```

Comment:
Hint: you can add useEffect with `errorMessage` in dependencies array
if it's not empty add 3000ms timeout and set `setErrorMessage('')`

It will allow to remove this in every method